src/olympia/users/models.py

```diff
@@ -479,27 +479,57 @@ def ban_and_disable_related_content_bulk(cls, users, move_files=False):
         from olympia.ratings.models import Rating
         from olympia.users.tasks import delete_photo
 
-        # collect affected addons
+        BannedUserContent.objects.bulk_create(
+            [BannedUserContent(user=user) for user in users], ignore_conflicts=True
+        )
+
+        # Collect affected addons
         addon_ids = set(
-            Addon.unfiltered.exclude(status=amo.STATUS_DELETED)
+            Addon.unfiltered.exclude(
+                status__in=(amo.STATUS_DELETED, amo.STATUS_DISABLED)
+            )
             .filter(addonuser__user__in=users)
             .values_list('id', flat=True)
         )
 
-        # First addons who have other authors we aren't banning
+        # First addons who have other authors we aren't banning - we are
+        # keeping the add-ons up, but soft-deleting the relationships.
         addon_joint_ids = set(
             AddonUser.objects.filter(addon_id__in=addon_ids)
             .exclude(user__in=users)
             .values_list('addon_id', flat=True)
         )
-        AddonUser.objects.filter(user__in=users, addon_id__in=addon_joint_ids).delete()
+        qs = AddonUser.objects.filter(user__in=users, addon_id__in=addon_joint_ids)
```

Comment:
nit: `qs` is an ambiguous variable name in a function that has many querysets.  And below as well. (unless this was intentional to avoid keeping many different querysets around longer than was needed?)

Reply:
Shouldn't make that much difference to keep the different querysets - I'll do that in https://github.com/mozilla/addons-server/pull/21472 which moves that entire method elsewhere